cmd/adapter/adapter.go

```diff
@@ -406,3 +410,16 @@ func makePrometheusCAClient(caFilePath string, tlsCertFilePath string, tlsKeyFil
 		},
 	}, nil
 }
+
+func parseHeaderArgs(args []string) map[string][]string {
```

Comment:
Can we use `http.Header` instead of `map[string][]string`? We could then add the headers via the `Add()` function of http.Headers.